VEDA_OS01/management/commands/tests/test_create_oauth_app_client.py

```diff
@@ -0,0 +1,185 @@
+"""
+Tests of the create_oauth_app_client management command.
+"""
+
+from itertools import product
+
+import ddt
+from django.contrib.auth import get_user_model
+from django.core.management import call_command
+from django.core.management.base import CommandError
+from django.test import TestCase
+from oauth2_provider.models import Application
+from VEDA_OS01.management.commands.create_oauth_app_client import CLIENT_TYPES, GRANT_TYPES
+
+
+USER_ID = 1
+USERNAME = 'username'
+CLIENT_ID = 'cliend_id101'
+REDIRECT_URI = 'https://www.example.com/o/token'
+CLIENT_TYPES = [client_type[0] for client_type in Application.CLIENT_TYPES]
+GRANT_TYPES = [client_type[0] for client_type in Application.GRANT_TYPES]
+
+# command line option names are used is Application model object creation. Some option
+# name does not directly transform to model field name, so replace them when needed.
+ALTERNATE_OPTION_NAME = {
+    'username': 'user_id',
+    'client_name': 'name',
+}
+
+
+@ddt.ddt
+class CreateOauthAppClientTests(TestCase):
+    """
+    Management command test class.
+    """
+    def setUp(self):
+        super(CreateOauthAppClientTests, self).setUp()
+        user_model = get_user_model()
+        self.user = user_model.objects.create(username=USERNAME)
+
+    def _call_command(self, args, options=None):
+        """
+        Call the command.
+        """
+        if options is None:
+            options = {}
+
+        call_command('create_oauth_app_client', *args, **options)
+
+    def assert_client_created(self, args, options):
+        """
+        Call the command and verify that the Client was created.
+        """
+        self._call_command(args, options)
```

Comment:
I think, only assertions should live here, this command invocation should be a part of the test cases.